dist/profile/templates/jenkinscontroller/casc/clouds-ec2.yaml.erb

```
@@ -22,6 +22,7 @@ jenkins:
       <%- $cloudInitDoneParentDir = "#{$tempDir}" -%>
       <%- $cloudInitDoneFile = "#{$cloudInitDoneParentDir}/.cloud-init.done" -%>
       <%- $javaHome = agent['javaHome'] ? agent['javaHome'] : @jcasc['agents_setup'][$os]['javaHome'] -%>
+      <%- $customDataDiskNotUsedForDocker = agent['customDataDiskNotUsedForDocker'] == nil ? false : agent['customDataDiskNotUsedForDocker'] -%>
```

In Ruby, you can test for both presence AND `true` value with `agent['customDataDiskNotUsedForDocker']`, like Groovy.

Given the logic order you used here, I suggest to use the same ternary order as the line above for `javaHome`



```suggestion
      <%- $customDataDiskNotUsedForDocker = agent['customDataDiskNotUsedForDocker'] ?  : false -%>
```

---

Even better (see comment below): the variable is unneeded as we only use it once.


```suggestion
```